### ISSUE TITLE
Add "turn off dpms/screensaver when game is running" function

### ISSUE THREAD

**Issue by RiantAnda** [lutris/lutris#2424]
This is a feature request, not an actual issue.

When I'm playing a game on lutris/wine using a controller, my screen turns off after 600 seconds, because DPMS/screensaver is kicking in. 
This does not happen for example when I use steam/proton. I'm assuming they automatically turn off dpms+screensavers.

It would be great to have a toggle under "System options" that disables dpms/screensaver when a game is running.

The commands behind the suggested function could be:
```
/usr/bin/xset s off -dpms     # to turn off dpms/screensaver
/usr/bin/xset s on +dpms   # enable dpms/screensaver
```

**Comment by tannisroot** [comment]
This annoyed the shit out of me when playing some games with my controller.
I am going to add this to 0.5.4 milestone because I fell like it's pretty a pretty important.
Note: this should be implemented as an option that's enabled by default and not as default and only behavior because we have installers for game launchers (Origin, Uplay, EGS, etc) and we don't to prevent screensaver/autosuspend when there's only a game launcher running and not an actual game.

**Comment by AlexanderRavenheart** [comment]
This feature should take into consideration to not enable dpms/screensaver if it was not enabled before game launch, it should save the state of the setting and restore it after game is closed, not blindly assume the user had the dpms/screensaver enabled.

**Comment by strycore** [comment]
I agree with @AlexanderRavenheart and also I'm not sure if the xset command is the universal way to set / unset the screensaver. We had a similar feature a while ago and had to be removed  because it wasn't reliable enough. Let's make sure it get properly implemented this time.

**Comment by tannisroot** [comment]
Perhaps source code of this Gnome Shell extension that disables suspend could help implement this properly: https://github.com/eonpatapon/gnome-shell-extension-caffeine

**Comment by RiantAnda** [comment]
I've learned today that Feral gamemode does screensaver inhibit by default.
The code is  here [https://github.com/FeralInteractive/gamemode/blob/master/daemon/gamemode-dbus.c](https://github.com/FeralInteractive/gamemode/blob/master/daemon/gamemode-dbus.c) at lines 706-783.
They are using a dbus function. Maybe this is already enough for devs to implement in Lutris?

Another interesting link discussing this topic/issue is here:
[https://stackoverflow.com/questions/460140/is-there-a-decent-way-to-inhibit-screensavers-in-linux](https://stackoverflow.com/questions/460140/is-there-a-decent-way-to-inhibit-screensavers-in-linux)

**Comment by guillaumezin** [comment]
I have these python scripts that work quite well, could be integrated in Lutris when launching a game:

```
#!/usr/bin/env python

import dbus
import time
import signal, sys

if len(sys.argv) != 3 :
        sys.stderr.write('Usage: pm-inhibit.py <appname> <reason>\n')
        sys.exit(1)

bus = dbus.Bus(dbus.Bus.TYPE_SESSION)
devobj = bus.get_object('org.freedesktop.ScreenSaver', '/org/freedesktop/ScreenSaver')
dev = dbus.Interface (devobj, "org.freedesktop.ScreenSaver")
cookie = dev.Inhibit(sys.argv[1], sys.argv[2])

while 1 == 1 :
        time.sleep(30)

dev.UnInhibit(cookie)
```

or

```
#!/usr/bin/env python

import dbus
import time
import signal, sys
import subprocess

if len(sys.argv) <= 3 :
        sys.stderr.write('Usage: pm-inhibit-then-launch.py <appname> <reason> <app> [<args>]\n')
        sys.exit(1)

bus = dbus.Bus(dbus.Bus.TYPE_SESSION)
devobj = bus.get_object('org.freedesktop.ScreenSaver', '/org/freedesktop/ScreenSaver')
dev = dbus.Interface (devobj, "org.freedesktop.ScreenSaver")
cookie = dev.Inhibit(sys.argv[1], sys.argv[2])

popen = subprocess.Popen(sys.argv[3:], stdout=subprocess.PIPE)
popen.wait()

dev.UnInhibit(cookie)
```
